### PR TITLE
remove discovery.zen.ping.unicast.hosts setting on esMajorVersion > 7

### PR DESCRIPTION
**Pull request by kuisathaverat**
it would fix https://github.com/elastic/helm-charts/issues/99 and https://github.com/elastic/helm-charts/issues/101

### CODE REVIEW

File: elasticsearch/templates/statefulset.yaml [hunk 1]
```diff
@@ -174,8 +174,13 @@ spec:
             value: "{{ .Values.minimumMasterNodes }}"
           {{- end }}
           {{- end }}
+          {{- if le (int .Values.esMajorVersion) 7 }}
           - name: discovery.zen.ping.unicast.hosts
             value: "{{ template "masterService" . }}-headless"
+          {{- else }}
```
Comment (Crazybus) [inline elasticsearch/templates/statefulset.yaml]: Could you add a test for this? You can more or less copy paste this existing one https://github.com/elastic/helm-charts/blob/c5289d5c3e78820d78baf0984c439c8ed68c893f/elasticsearch/tests/elasticsearch_test.py#L230-L246

File: elasticsearch/templates/statefulset.yaml [hunk 2]
```diff
@@ -174,8 +174,13 @@ spec:
             value: "{{ .Values.minimumMasterNodes }}"
           {{- end }}
           {{- end }}
+          {{- if le (int .Values.esMajorVersion) 7 }}
```
[the remark this answers is not in the document]
Comment (kuisathaverat) [reply elasticsearch/templates/statefulset.yaml]: nop, we can make it and remove the warning.

File: elasticsearch/templates/statefulset.yaml (hunk 1 above)
Comment (kuisathaverat) [reply elasticsearch/templates/statefulset.yaml]: sure

**Comment by Crazybus** [comment]
jenkins test this please

**Pull request merged**